R/spec.R

```diff
@@ -119,7 +119,11 @@ spec_reason <- function(call, signaller) {
   )
 }
 
-spec_package <- function(env, signaller) {
+spec_pkg <- function(pkg, env, signaller) {
+  if (!is_null(pkg) || is.null(env)) {
+    return(pkg)
```

Comment:
So the idea is that if both `pkg` and `env` are `NULL`, then this will return `NULL` (rather than a character vector of size 1) but the user of that result should just never look at `pkg` when generating the message

Could also return `"<NA>"` in the case where `pkg` and `env` are both `NULL` for type stability

Reply:
I think `NULL` is more convenient. For instance there is no predicate for `"<NA>"`, nor `%||%` support.